sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -422,6 +432,7 @@ class StreamExecution(
     val replacementMap = AttributeMap(replacements)
     val triggerLogicalPlan = withNewSources transformAllExpressions {
       case a: Attribute if replacementMap.contains(a) => replacementMap(a)
+      case t: CurrentTimestamp => CurrentBatchTimestamp(currentBatchTimestamp)
```

Comment:
Nevermind, I just read the JIRA, and I think I understand what is going on.

To me the `time` column in `df.groupBy('time)` is different from the `time` column which is  (apparently) in the source. I personally think it is better to have the statefull aggregate produce a new time column, then to restrict push down.